File: CSharp/leetcodesln/SumOfSquareNumbers.cs
using System;

namespace leetcodesln
{
    public class SumOfSquareNumbers
    {
        public bool JudgeSquareSum(int c)
        {
            int l = 0, r = (int)Math.Sqrt(c);
            while (l <= r)
            {
                if (l * l + r * r == c) return true;
                if (l * l + r * r < c) l++;
                if (l * l + r * r > c) r--;

            }
            return false;
        }
    }
}

File: Go/questions/SumOfSquareNumbers.go
package questions

import "math"

func JudgeSquareSum(c int) bool {
	sqrt := math.Sqrt(float64(c))
	return canFind(sqrt, c)
}

func canFind(f float64, c int) bool {

	l, r := 0, int(f)

	for l <= r {
		if l*l+r*r == c {
			return true
		} else if l*l+r*r > c {
			r--
		} else {
			l++
		}
	}
	return false
}
